src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp

```diff
@@ -0,0 +1,250 @@
+  addParameter( appendGeorefParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > exportRDFParam = qgis::make_unique< QgsProcessingParameterBoolean >( QStringLiteral( "INCLUDE_METADATA" ), QObject::tr( "Export RDF metadata (title, author, etc.)" ), true );
+  exportRDFParam->setFlags( exportRDFParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( exportRDFParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > antialias = qgis::make_unique< QgsProcessingParameterBoolean >( QStringLiteral( "ANTIALIAS" ), QObject::tr( "Enable antialiasing" ), true );
+  antialias->setFlags( antialias->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( antialias.release() );
+}
+
+QgsProcessingAlgorithm::Flags QgsLayoutAtlasToImageAlgorithm::flags() const
+{
+  return QgsProcessingAlgorithm::flags() | FlagNoThreading;
+}
+
+QgsLayoutAtlasToImageAlgorithm *QgsLayoutAtlasToImageAlgorithm::createInstance() const
+{
+  return new QgsLayoutAtlasToImageAlgorithm();
+}
+
+QVariantMap QgsLayoutAtlasToImageAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  // this needs to be done in main thread, layouts are not thread safe
+  QgsPrintLayout *layout = parameterAsLayout( parameters, QStringLiteral( "LAYOUT" ), context );
+  if ( !layout )
+    throw QgsProcessingException( QObject::tr( "Cannot find layout with name \"%1\"" ).arg( parameters.value( QStringLiteral( "LAYOUT" ) ).toString() ) );
+
+  QString error;
+  QgsLayoutAtlas *atlas = layout->atlas();
+
+  QgsVectorLayer *previousCoverageLayer = atlas->coverageLayer();
+  const bool previousEnabled = atlas->enabled();
+  const QString previousFilenameExpression = atlas->filenameExpression();
+  const QString previousFilterExpression = atlas->filterExpression();
+  const bool previousSortFeatures = atlas->sortFeatures();
+  const bool previousSortAscending = atlas->sortAscending();
+  const QString previousSortExpression = atlas->sortExpression();
+
+  auto restoreAtlas = [ atlas,
+                        previousCoverageLayer,
+                        previousEnabled,
+                        previousFilenameExpression,
+                        previousFilterExpression,
+                        previousSortFeatures,
+                        previousSortAscending,
+                        previousSortExpression ]()
+  {
+    QString error;
+    atlas->setEnabled( previousEnabled );
+    atlas->setCoverageLayer( previousCoverageLayer );
+    atlas->setFilenameExpression( previousFilenameExpression, error );
+    atlas->setFilterExpression( previousFilterExpression, error );
+    atlas->setSortFeatures( previousSortFeatures );
+    atlas->setSortAscending( previousSortAscending );
+    atlas->setSortExpression( previousSortExpression );
+  };
+
+  QgsVectorLayer *layer = parameterAsVectorLayer( parameters, QStringLiteral( "COVERAGE_LAYER" ), context );
```

Comment:
What about making the coverage layer parameter (and other atlas settings) optional, and using their existing values if not set? That way you could use this algorithm to export an existing atlas which is already fully setup in the layout designer.

Reply:
How about this: if a coverage layer is not provided, the algorithm exports whatever is already setup in the designer (and throws an error if a layout doesn't have an enabled atlas). If a coverage layer is provided, then all settings are overwritten using the parameters. 

The need to overwrite everything arises from the fact that we currently thread empty parameters (eg sort by expression) as a way to tell the algorithm to deactivate sorting.

Reply:
Sounds good! Just make sure the docs explain this behaviour